flagscale/backends/vllm/vllm/model_executor/models/minicpmv.py

```diff
@@ -0,0 +1,1433 @@
+# SPDX-License-Identifier: Apache-2.0
```

Comment:
At top line, add this comment:
Copied adopted [from https://github.com/vllm-project/vllm/blob/eccdc8318c50ad3e2adfaaf99d714a8dccc6ab16/vllm/distributed/kv_transfer/kv_connector/v1/p2p/p2p_nccl_engine.py](https://github.com/vllm-project/vllm/pull/22166/files#diff-7b8a4e258637b7c94389c745c449c52137d33cf92957f3e5bcb18a0ee204b21b)
Below is the original copyright: